src/clarity-icons/clarity-icons-element.ts

```diff
@@ -20,28 +22,30 @@ export function ClarityIconElement() {
     "use strict";
     return (parentConstructor as any).apply(this, arguments);
 }
-
-(ClarityIconElement as any).observedAttributes = ["shape", "size"];
+(ClarityIconElement as any).observedAttributes = ["shape", "size", "title"];
 
 ClarityIconElement.prototype = Object.create(
     HTMLElement.prototype, {constructor: {configurable: true, writable: true, value: ClarityIconElement}});
 
 ClarityIconElement.prototype.constructor = ClarityIconElement;
 
-const generateIcon = function(element: any, shape: string) {
-    shape = shape.split(/\s/)[0];
+const replaceTitleWithCustom = function(currentShapeTemplate: string, currentTitleAttrVal: string,
+                                        clrIconUniqId: string) {
+    const hasSVGtag = currentShapeTemplate.match(/<svg\b[^>]*>/i);  // will find the first occurrence of svg tag
```

Comment:
Do you need a regex here? `indexOf()` with plain strings is way faster than regex matching, and I think matching `"<svg"` is enough, isn't it?

Reply:
@youdz I've updated replaceTitleWithCustom function. This updated version doesn't rely on the regex; it doesn't remove elements such as title from user's template. It converts the template into a dom element and makes all the required changes into it before injecting it into the icon.